ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/KnownPeers.hs

```diff
@@ -176,9 +183,14 @@ jobPeerShare PeerSelectionActions{requestPeerShare}
                -- which means other smaller valency nodes won't be shared that
                -- much. So any chance we can have of filtering them from peer
                -- sharing results is welcome.
+               --
+               -- We also filter ourselves from the result set.
                newPeers    = [ p | Right (PeerSharingResult ps) <- totalResults
                                  , p <- ps
-                                 , not (isKnownLedgerPeer p (knownPeers st)) ]
+                                 , not (isKnownLedgerPeer p (knownPeers st))
+                                 , Just p /= mbLocalIPv4
```

Comment:
I'm not certain this will give the behaviour you want under various deployment scenarios. 

It will "work" for a single connected host directly with a public IP address, in other scenarios loopback connections may well be established.

What is the consequence of creating a loopback connection?